R/pipeline.R

```diff
@@ -71,23 +71,32 @@
 orchestrate_pipeline <- function(config_path,
                                  blob_storage_container = NULL,
                                  output_dir = "/") {
-  # TODO: Add config reader here
-  config <- jsonlite::read_json(config_path,
-    simplifyVector = TRUE
+  config <- rlang::try_fetch(
+    read_json_into_config(config_path, c("exclusions")),
+    error = function(con) {
+      cli::cli_warn("Bad config file",
+        parent = con,
+        class = "Bad_config"
+      )
+      FALSE
+    }
```

Comment:
Just confirming that this will throw the error message too, and that the error message will have more details about what was bad? (I think this was one of the main motivations for creating a class)

Reply:
Yes, this will cause the whole pipeline to stop, and it will print out what fields are missing from the config file. Here's an example from using a config file that is missing most of the fields.

```R
r$> pipeline_success <- orchestrate_pipeline(
  config_path = config_path,
  blob_storage_container = NULL,
  output_dir = output_dir
)
ℹ Optional field not in config file: `exclusions`
Warning message:
Bad config file
Caused by error in `read_json_into_config()` at cfa-epinow2-pipeline/R/pipeline.R:74:3:
! Properties not in the config file: `min_reference_date`, `max_reference_date`,
  `report_date`, `disease`, `geo_value`, `geo_type`, `seed`, `horizon`, `model`,
  `config_version`, `quantile_width`, `data`, `priors`, `parameters`, and
  `sampler_opts`
```